relay_test.go

```diff
@@ -404,6 +404,11 @@ func TestLargeTimeoutsAreClamped(t *testing.T) {
 			close(done)
 		}()
 
+		// This test is very sensitive to system noise, where a spike of latency in the relay (e.g. caused by load)
+		// is able to cause the client call to timeout. As such, testing this with an end-to-end call is prone to
+		// false negatives.
+		// TODO: consider redesigning the test to be more "white-box", e.g. by checking the TTL field of the frame
+		// in a second relay
```

Comment:
we're effectively doing that with the server-side timeout, so not sure a second relay adds much value?

Reply:
what i had in mind was to check the clamped TTL at the frame level, i.e. what the relay sets in
```
	ttl := f.TTL()
	if ttl > r.maxTimeout {
		ttl = r.maxTimeout
		f.SetTTL(r.maxTimeout)
	}
```
supposedly if we go through a MITM relay, we should be able to inspect the updated TTL there?

Reply:
that would probably be separate from verifying that the relay correctly times out the call though, so we'd probably still have to deal with the flakiness from and end-to-end test :(

Reply:
ah, I didn't realize that the deadline check was already doing this, my bad. please ignore 🤦